src/vs/workbench/api/browser/mainThreadEditors.ts

```diff
@@ -116,6 +127,75 @@ export class MainThreadTextEditors implements MainThreadTextEditorsShape {
 		return result;
 	}
 
+	private _getTextEditorDiffInformation(textEditor: MainThreadTextEditor): IObservable<ITextEditorDiffInformation | undefined> {
+		const codeEditor = textEditor.getCodeEditor();
+		if (!codeEditor) {
+			return constObservable(undefined);
```

Comment:
I wonder if this is a deadend in case there is no editor? It won't ever recover and never send diff info

Reply:
We do call `setCodeEditor()` in the constructor but I do see a case in which we reset the code editor if the model has changed. I do not know enough about this world to know whether this would be a real concern. Worst case scenario would be that we would not be sending diff information unless I am missing something. https://github.com/microsoft/vscode/blob/b83c4f6a42bb78d6737ac097756fb05e7cd99f99/src/vs/workbench/api/browser/mainThreadEditor.ts#L255